src/mlpack/core/data/load_csv.hpp

```diff
@@ -119,33 +205,55 @@ class LoadCSV
     while (std::getline(inFile, line))
     {
       ++rows;
-      // Remove whitespace from either side.
-      boost::trim(line);
-
+      // Remove whitespaces from either side
+      trim(line);
       if (rows == 1)
       {
         // Extract the number of columns.
-        auto findColSize = [&cols](iter_type) { ++cols; };
-        qi::parse(line.begin(), line.end(),
-            stringRule[findColSize] % delimiterRule);
+        std::pair<size_t, size_t> dimen = GetNonNumericMatSize(inFile, delim);
+        cols = dimen.second;
       }
 
       // I guess this is technically a second pass, but that's ok... still the
       // same idea...
       if (MapPolicy::NeedsFirstPass)
       {
         // In this case we must pass everything we parse to the MapPolicy.
-        auto firstPassMap = [&](const iter_type& iter)
+        std::string str(line.begin(), line.end());
+
+        std::stringstream line_stream;
```

Comment:
Small style issue; can you call this `lineStream`?  I know it is a small thing, but I think consistency throughout the codebase (even with respect to variable names) is important for maintainability and readability.